pkg/oci/registry/client.go

```diff
@@ -116,10 +116,21 @@ func (c *Client) GetImageVersion(ctx context.Context, imageName string) (ImageVe
 	if err != nil {
 		return ImageVersion{}, fmt.Errorf("descriptor.Image(): %w", err)
 	}
-	dig, err := img.Digest()
+
+	// use image digest as a fallback
+	digestFn := img.Digest
```

Comment:
was it on purpose to not use the Digest() function?

Reply:
I'm using a function pointer here that is later on used to call the Digest() function. Depending on the outcome of line 124, the function pointer might be switched to a different Digest() function. I tried to avoid lengthy if... else constructs here. Alternative implementation:
```
	var dig containerv1.Hash
	// try to get image manifest to cover multi arch images
	if imgIdx, err := descriptor.ImageIndex(); err == nil {
		dig, err = imgIdx.Digest()
		if err != nil {
			return ImageVersion{}, fmt.Errorf("img.Digest(): %w", err)
		}
	} else {
		dig, err = img.Digest()
		if err != nil {
			return ImageVersion{}, fmt.Errorf("img.Digest(): %w", err)
		}
	}
```